tasks/Task.cpp

```diff
@@ -13,37 +18,666 @@ Task::~Task()
+    // to gain control If there is no switchover key, or an incorrect switchover key,
+    // ReadWrite access will not be granted
+    Arena::SetNodeValue<int64_t>(m_device->GetTLDeviceNodeMap(),
+        "CcpSwitchoverKey",
+        m_switchover_key);
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetTLDeviceNodeMap(),
+        "DeviceAccessStatus",
+        "ReadWrite");
+
+    // confirm if operation was successfull
+    device_access_status =
+        Arena::GetNodeValue<GenICam::gcstring>(m_device->GetTLDeviceNodeMap(),
+            "DeviceAccessStatus");
+    if (device_access_status == "ReadWrite") {
+        LOG_INFO_S << "Task retrieved the Camera Read/Write Access." << endl;
+        return true;
+    }
+
+    LOG_WARN_S << "Task failed to retrieved the Camera Read/Write Access." << endl;
+    return false;
+}
+
+bool Task::configureCamera()
+{
+    LOG_INFO_S << "Performing Factory Reset" << endl;
+    if (!factoryReset()) {
+        return false;
+    }
+
+    LOG_INFO_S << "Configuring camera." << endl;
+    if (!acquisitionConfiguration()) {
+        return false;
+    }
+
+    LOG_INFO_S << "Setting StreamBufferHandlingMode." << endl;
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetTLStreamNodeMap(),
+        "StreamBufferHandlingMode",
+        "NewestOnly");
+
+    LOG_INFO_S << "Setting StreamAutoNegotiatePacketSize." << endl;
+    Arena::SetNodeValue<bool>(m_device->GetTLStreamNodeMap(),
+        "StreamAutoNegotiatePacketSize",
+        true);
+
+    LOG_INFO_S << "Setting StreamPacketResendEnable." << endl;
+    Arena::SetNodeValue<bool>(m_device->GetTLStreamNodeMap(),
+        "StreamPacketResendEnable",
+        true);
+
+    LOG_INFO_S << "Setting PixelFormat." << endl;
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetNodeMap(),
+        "PixelFormat",
+        convertFrameModeToPixelFormat(_image_config.get().format,
+            _image_config.get().depth)
+            .c_str());
+
+    // When horizontal binning is used, horizontal decimation (if supported) is not
+    // available. When vertical binning is used, vertical decimation (if supported) is not
+    // available.
+
+    // If the camera is acquiring images, AcquisitionStop must be called before adjusting
+    // binning settings.
+    if (!binningConfiguration()) {
+        return false;
+    }
+
+    if (!decimationConfiguration()) {
+        return false;
+    }
+
+    if (!dimensionsConfiguration()) {
+        return false;
+    }
+
+    if (!exposureConfiguration()) {
+        return false;
+    }
+
+    Frame* frame = new Frame(_image_config.get().width,
+        _image_config.get().height,
+        _image_config.get().depth,
+        _image_config.get().format,
+        0,
+        0);
+    m_frame.reset(frame);
+
+    return true;
+}
+
+void Task::startCamera()
+{
+    LOG_INFO_S << "Starting frame acquisition." << endl;
+    m_device->StartStream();
+}
+
+void Task::acquireFrame()
+{
+    try {
+        Arena::IImage* image =
```

Comment:
This code is not exception-safe because of this very pointer.

Create a small struct that manages the image RAII-style, to auto-requeue it.